src/news_gui.cpp

```diff
@@ -311,7 +311,8 @@ struct NewsWindow : Window {
 		/* Initialize viewport if it exists. */
 		NWidgetViewport *nvp = this->GetWidget<NWidgetViewport>(WID_N_VIEWPORT);
 		if (nvp != nullptr) {
-			nvp->InitializeViewport(this, ni->reftype1 == NR_VEHICLE ? 0x80000000 | ni->ref1 : (uint32)GetReferenceTile(ni->reftype1, ni->ref1), ZOOM_LVL_NEWS);
+			auto zoom_level = ZoomLevel(ZOOM_LVL_NEWS + _gui_zoom - ZOOM_LVL_OUT_4X); // adjust zoom level for GUI scaling (HiDPI)
```

Comment:
I probably should introduce a helper method for adjusting the zoom level, especially as it's been repeated a few times for every viewport flavour.